src/server/cluster/cluster_family.cc

```diff
@@ -401,11 +409,15 @@ void ClusterFamily::ReadWrite(CmdArgList args, ConnectionContext* cntx) {
 }
 
 void ClusterFamily::DflyCluster(CmdArgList args, ConnectionContext* cntx) {
-  if (!IsClusterEnabledOrEmulated()) {
-    return cntx->SendError(kClusterDisabled);
+  if (!(IsClusterEnabled() || (IsClusterEmulated() && cntx->journal_emulated))) {
+    return cntx->SendError("Cluster is disabled. Use --cluster_mode=yes to enable.");
   }
 
-  VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  if (cntx->conn()) {
+    VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  } else {
+    VLOG(2) << "Got DFLYCLUSTER command (NO_CLIENT_ID): " << args;
+  }
```

Comment:
```suggestion
  auto print_conn = [](auto* conn) { return conn ? conn->GetClientId() : "NO_CLIENT_ID" };
  VLOG(2) << "Got DFLYCLUSTER command (" << print_conn(cntx->conn()) << "): " << args;
```

Reply:
it's different types. int and char[]

Reply:
You could return a string.. no need to duplicate code
(a string < 16 bytes will not do dynamic allocations)

Reply:
Will change it next time when do some changes in this file